pkg/operator/controller/ingress/deployment.go

```diff
@@ -430,6 +430,19 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		env = append(env, corev1.EnvVar{Name: "ROUTER_IP_V4_V6_MODE", Value: mode})
 	}
 
+	routeAdmission := operatorv1.RouteAdmissionPolicy{
+		NamespaceOwnership: operatorv1.StrictNamespaceOwnershipCheck,
+	}
+	if admission := ci.Spec.RouteAdmission; admission != nil && len(admission.NamespaceOwnership) > 0 {
+		routeAdmission.NamespaceOwnership = ci.Spec.RouteAdmission.NamespaceOwnership
+	}
+	switch routeAdmission.NamespaceOwnership {
+	case operatorv1.StrictNamespaceOwnershipCheck:
+		env = append(env, corev1.EnvVar{Name: "ROUTER_DISABLE_NAMESPACE_OWNERSHIP_CHECK", Value: "false"})
```

Comment:
Alternatively, because the default is false, we could omit the environment variable and thereby avoid triggering a rolling update on old clusters.  That's what I did with `ROUTER_IP_V4_V6_MODE`:  https://github.com/openshift/cluster-ingress-operator/blob/a11c8f1d9a7e4cc2334e959700c1257015f5cf23/pkg/operator/controller/ingress/deployment.go#L425-L431
On the other hand, we will probably get a rolling update anyway because we are updating the router image, so maybe it does not matter anyway.

Reply:
I had the same thoughts, can't decide if adding code to avoid the rollout is worth it.

Reply:
s/adding/omitting/

I'm suggesting we modify `env` only in the `operatorv1.InterNamespaceAllowedOwnershipCheck` case.

Reply:
Thought about this some more.

1. As we already noted, if you're running this new operator code, we'll rollout out a new router deployment anyway for the new image version, so that part seems like a wash
2. I think it's safer generally to explicitly set the variable in all cases to true or false to avoid relying on defaulting in the router in the Strict case

So I would suggest what I have here is good. WDYT?

Reply:
Agreed, explicit is good, and we anticipate a new router image, so might as well always add it.